frontend/src/components/FundsPullPreApproval/ApprovalDetails.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useContext } from "react";
+import { Approval } from "../../interfaces/approval";
+import { diemAmountToHumanFriendly } from "../../utils/amount-precision";
+import { settingsContext } from "../../contexts/app";
+
+interface RequestDetailsProps {
+  approval: Approval | undefined;
+}
+
+function ApprovalDetails({ approval }: RequestDetailsProps) {
+  const [settings] = useContext(settingsContext)!;
+
+  const dateToDisplayByApprovalStatus = () => {
+    if (approval!.status === "pending") {
+      return (
+        <>
+          {"on"} <span>{new Date(approval!.created_at).toLocaleString()}</span>
+        </>
+      );
+    }
+    if (approval!.status === "rejected") {
+      return (
+        <div>
+          {"Rejected on"} <span>{new Date(approval!.updated_at).toLocaleString()}</span>
```

Comment:
why we hold rejection timestamp on updated at? shouldn't there be a separate field for this?

Reply:
@vladkucherov - To avoid make a column to each status I'm using the updated_at to the rejection and the revoke, I assume no update will happen after this statuses been set. The "approve" received dedicated column\field (all this data actually arrive from the DB through api calls) since we want to display his value in all cases